api/app/db/VersionsDao.scala

```diff
@@ -241,6 +252,30 @@ class VersionsDao @Inject() (
     }
   }
 
+  def findAllByVersionFilter(
```

Comment:
I'm wondering if we could always use `and(s"versions.version like '${versionConstraint}%'")` (when not None) in the existing `findAll` function instead of having to duplicate much of the logic. For version filters like `x.y.z` it would just still select only that version.

Reply:
Also, it should be relatively easy to write a test for this feature in `VersionsDaoSpec`.

Reply:
Both done